src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParametersUpdatedHandler.cs

```diff
@@ -0,0 +1,58 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Fhir.Core.Features.Search.Registry;
+using Microsoft.Health.Fhir.Core.Messages.Reindex;
+using Microsoft.Health.Fhir.Core.Messages.Search;
+
+namespace Microsoft.Health.Fhir.Core.Features.Definition
+{
+    public class SearchParametersUpdatedHandler : INotificationHandler<SearchParametersHashUpdated>, INotificationHandler<ReindexJobCompleted>
+    {
+        private SearchParameterDefinitionManager _searchParameterDefinitionManager;
+        private readonly ISearchParameterRegistry _searchParameterRegistry;
+
+        public SearchParametersUpdatedHandler(SearchParameterDefinitionManager searchParameterDefinitionManager, ISearchParameterRegistry searchParameterRegsitry)
+        {
+            EnsureArg.IsNotNull(searchParameterDefinitionManager, nameof(searchParameterDefinitionManager));
+            EnsureArg.IsNotNull(searchParameterDefinitionManager, nameof(searchParameterDefinitionManager));
+
+            _searchParameterDefinitionManager = searchParameterDefinitionManager;
+            _searchParameterRegistry = searchParameterRegsitry;
+        }
+
+        public Task Handle(SearchParametersHashUpdated notification, CancellationToken cancellationToken)
+        {
+            _searchParameterDefinitionManager.SearchParametersHash = notification.HashValue;
+            return Task.CompletedTask;
+        }
+
+        public Task Handle(ReindexJobCompleted notification, CancellationToken cancellationToken)
+        {
+            var searchParameterStatusList = new List<ResourceSearchParameterStatus>();
+            foreach (string uri in notification.SearchParameterUrls)
+            {
+                var searchParamUri = new Uri(uri);
+                var searchParam = _searchParameterDefinitionManager.GetSearchParameter(searchParamUri);
```

Comment:
nit: Using a Get call in order to update the `SearchParameterInfo` value in `SearchParameterDefinitionManager` seems a little unintuitive. And can cause issues in the future when multiple callers are using this at the same time. Maybe we should add an API call for updating the search param info?

Reply:
I will add an API.